### PR TITLE
Add a tool in record set dialog to merge RS together

### LINKED ISSUE #1665: Add ability to merge record sets
Add ability to merge record sets

Suggested by @tlammer

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #1665

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add automated tests
- [ ] Add relevant issue to release milestone

### Testing instructions

- open record set side bar menu 
- click button merge 
- select min 2 RS 
- [ ] verify you can only select RS that are from the same table 
- click merge 
- wait for the page to reload 
- [ ] verify in the record set side bar menu that the RS have been merged, a new one has been created and the merged ones have been deleted. The new RS has the name of the first or one of the selected RS.

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/Toolbar/MergeRecordSets.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          header={treeText.merge()}
+          onClose={handleClose}
+        >
+          <Ul>
+            {recordSets?.map((recordSet) => (
+              <li key={recordSet.id}>
+                <Label.Inline>
+                  <Input.Checkbox
+                    checked={selectedRecordSets.includes(recordSet.id)}
+                    onValueChange={(): void => onSelected(recordSet)}
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Toolbar/MergeRecordSets.tsx:117]: minor idea: once at least on record set is selected, render all checkboxes for record sets from different base table as disabled?

**Comment by CarolineDenis** [comment]
TODO: 
Combine this component with the Add Record Set toll (add CO in accession)

**Comment by CarolineDenis** [comment]
NOTES: 
At the moment the new RS have this nomination: {TableName} Merged Recordset {YEAR-MONTH-DAY} 
It's possible that the user will have several RS with the same name if they merge multiple times in a day. 
The user will have to rename their RS appropriately. 
Or we can add the time too.

**Review by alesan99 — APPROVED** [review]
### Testing instructions

- open record set side bar menu 
- click button merge 
- select min 2 RS 
- [x] verify you can only select RS that are from the same table 
- click merge 
- wait for the page to reload 
- [x] verify in the record set side bar menu that the RS have been merged, a new one has been created and the merged ones have been deleted. The new RS has the name of the first or one of the selected RS. 

Tested many record sets in different contexts, working as intended 👍👍👍

**Comment by CarolineDenis** [comment]
@emenslin 
Yes it normal behavior, if you want all the RS you can select 'Unlimited' in the bottom right dropdown to chnage the limit of RS

**Pull request merged**